tracing-tracy/src/lib.rs

```diff
@@ -99,12 +100,14 @@ impl<F> TracyLayer<F> {
     /// Note that enabling callstack collection can and will introduce a non-trivial overhead at
     /// every instrumentation point. Specifying 0 frames (which is the default) will disable stack
     /// trace collection.
-    pub fn with_stackdepth(mut self, stack_depth: u16) -> Self {
+    #[must_use]
+    pub const fn with_stackdepth(mut self, stack_depth: u16) -> Self {
         self.stack_depth = stack_depth;
         self
     }
 
     /// Use a custom field formatting implementation.
+    #[must_use]
     pub fn with_formatter<Fmt>(self, fmt: Fmt) -> TracyLayer<Fmt> {
```

Comment:
This can be a `const fn` too, can’t it?

Reply:
Interestingly enough, no because self gets dropped:
```
destructor of `TracyLayer<F>` cannot be evaluated at compile-time
```